smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -339,12 +337,16 @@ contract MixinKeys is
     uint _tokenId
   ) internal
   {
-    if (ownerOf(_tokenId) != _keyOwner) {
-      // TODO: this may include duplicate entries
+
+    // check expiration ts should be set to know if owner had previously registered a key 
+    Key memory key = keyByOwner[_keyOwner];
+    if(key.expirationTimestamp == 0 ) {
```

Comment:
when reseting a key, we set tokenId to 0 and expTs to block.timestamp. So we can check if a user key has an existing block.ts to see if he had purchased a key in the past